extensions/amp-ad-network-cloudflare-impl/0.1/amp-ad-network-cloudflare-impl.js

```diff
@@ -62,18 +49,44 @@ export class AmpAdNetworkCloudflareImpl extends AmpA4A {
     return ['cloudflare'];
   }
 
+
+
   /** @override */
   getAdUrl() {
-    // You can optionally use data from this context to augment the URL,
-    // to pass data back to the network, to aid in ad selection.  For example:
-    //   const rect = this.getIntersectionElementLayoutBox();
-    // can extract the layout size and be used to pass layout hints to the server.
+    const rect = this.getIntersectionElementLayoutBox();
+    const elem = this.element;
 
-    // If you don't want to use relative URL generation from your base url,
-    // be sure that this only generates urls to valid hosts, since it could
-    // allow cookie leaks, etc.
+    // handle variable replacements
+    function replacements(str) {
+      if (!str) {
+        return str;
+      }
+
+      // allow injection of width and height as parameters
+      str = str.replace(/{width}/g, (rect.width || 0).toString());
+      str = str.replace(/{height}/g, (rect.height || 0).toString());
+
+      return str;
+    }
+
+    let url = this.element.getAttribute('src');
```

Comment:
We'd like to have some kind of whitelist of supported ad networks. That would mean a trivial PR every time you onboard a customer.

Have you considered doing something like the vendors.js file used by amp-analytics? I think the current design would require retagging publisher sites every time a network decides it would like to change their ad request URL.